### PR TITLE
Remove vendor directory

### PR DESCRIPTION
**Pull request by baude**
From now on, the vendor directory will no longer be part of the github
repository.  For distribution maintainers, we will publish the vendor
directory in the form of a tar.gz file with the release.

For the Fedora spec, something like this will to be done at a minimum:

+Source1: %{url}/releases/download/v%{version}/%{name}-%{version}-vendor.tar.gz

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
review and merge please.

@lsm5 PTAL

**Comment by lsm5** [comment]
@baude how will the vendored tarball be generated? Does github publish that automatically from `make vendor` or will that be a manual step?

**Comment by baude** [comment]
@lsm, make vendor and tar that up ... we will do it with releases.

**Review by lsm5 — APPROVED** [review]
LGTM

File: contrib/cirrus/runner.sh [hunk 1]
```diff
@@ -19,6 +19,7 @@ _run_noarg() {
 _run_build() {
     # Assume we're on a fast VM, compile everything needed by the
     # rest of CI since subsequent tasks may have limited resources.
+	make vendor
```
Comment (mheon) [inline contrib/cirrus/runner.sh:22]: Do we need these? Shouldn't Cargo automatically fetch deps?

Comment (baude) [reply contrib/cirrus/runner.sh:22]: they were needed or the sources were not present.

**Pull request merged**